src/components/Table/useColumns.ts

```diff
@@ -0,0 +1,21 @@
+import { Dispatch, SetStateAction, useMemo, useState } from "react";
+import { GridColumn, Kinded } from "./GridTable";
+
+export interface Columns<R extends Kinded, S = {}> {
```

Comment:
**nit/suggestion:** My gut tells me this is more of a `type` than an `interface`. Typically we use `interface`s as the definition of what props are required for "interfacing" with a component. Where a type is more the structure of an object. Can we update this to be a `type` instead of an interface, as it is the return type of the memo'd data.